src/pylogctx/django.py

```diff
@@ -4,19 +4,19 @@
 
 from django.conf import settings
 
-from pylogctx import context
+from pylogctx import context, AdapterNotFound
 
 
 logger = logging.getLogger(__name__)
 
 
-class ExtractRequestContextMiddleware(object):
+class OuterMiddleware(object):
     def process_request(self, request):
-        extractor = settings.PYLOGCTX_REQUEST_EXTRACTOR
+        context.clear()
         try:
-            context.update(**extractor(request))
-        except Exception:
-            logger.exception()
+            context.update(request)
+        except AdapterNotFound:
+            logger.debug("Can't adapt %s for log.", request.__class__.__name__)
```

Comment:
Right. But I would like to make it optionnal to register a request adapter. E.g. you just want the outermiddleware to clear the context.

So i would vote for info.